src/pybind/mgr/ansible/module.py

```diff
@@ -179,18 +186,77 @@ def is_effective(self):
         In the case of Ansible, this will be True if the playbooks has been
         executed succesfully.
 
-        :return Boolean: if the playbook has been executed succesfully
+        :return Boolean: if the playbook/ARS operation has been executed
+                         succesfully
         """
 
-        return self.status == ExecutionStatusCode.SUCCESS
+        return self._status == ExecutionStatusCode.SUCCESS
 
     @property
     def is_errored(self):
-        return self.error
+        return self._status == ExecutionStatusCode.ERROR
 
     @property
-    def is_complete(self):
-        return self.is_errored or (self.is_persistent and self.is_effective)
+    def result(self):
+        return self._result
+
+
+class ARSChangeOperation(AnsibleChangeOperation):
+    """Execute an Ansible Runner Service Operation that implies a change in
+    the cluster
+    """
+    def __init__(self, client, logger, url, http_operation, payload={}):
+        """
+        :param client         : Ansible Runner Service Client
+        :param logger         : The object used to log messages
+        :param url            : The Ansible Runner Service URL that provides
+                               the operation
+        :param http_operation : Type of http operation (post,delete,put)
+        :param payload        : http request payload
+        """
+        super(ARSChangeOperation, self).__init__()
+
+        self.ar_client = client
+        self.log = logger
+        self.url = url
+        self.http_operation = http_operation
+        self.payload = payload
+
+        self.process_output = None
+
+    def __str__(self):
+             return "Ansible Runner Service: {operation} {url}".format(
+                operation = self.http_operation,
+                url = self.url)
+
+    @property
+    def status(self):
+        """Execute the Ansible Runner Service operation and update the status
+        and result of the underlying Completion object.
+        """
+
+        # Execute the right kind of http request
+        try:
+            if self.http_operation == "post":
+                response = self.ar_client.http_post(self.url, self.payload)
+            elif self.http_operation == "delete":
+                response = self.ar_client.http_delete(self.url)
+        # Any kind of error communicating with ARS or getting
+        # an empty http request
+        except Exception as ex:
+            self._status = ExecutionStatusCode.ERROR
+            self._result = ex
```

Comment:
`self._result` should be either of type `Exception` or `str`. E.g. by forcing it to be `str`:

```suggestion
            self._result = str(ex)
```

Reply:
True. Fixed. Thx!